agent/otel/otlpmqttexporter/otlp.go

```diff
@@ -95,18 +108,69 @@ func (e *exporter) start(_ context.Context, _ component.Host) error {
 	return nil
 }
 
+// extractAttribute extract attribute from metricsRequest metrics
+func (e *exporter) extractAttribute(metricsRequest pmetricotlp.Request, attribute string) string {
+	metrics := metricsRequest.Metrics().ResourceMetrics().At(0).ScopeMetrics().At(0).Metrics()
+	for i := 0; i < metrics.Len(); i++ {
+		metricItem := metrics.At(i)
+		if metricItem.Name() == "dns_wire_packets_tcp" || metricItem.Name() == "packets_ipv4" || metricItem.Name() == "dhcp_wire_packets_ack" || metricItem.Name() == "flow_in_udp_bytes" {
```

Comment:
hm we can't hard code metric names in here. i guess the problem is we otherwise don't know how to pull out the attribute, because the metric type is dynamic?

Reply:
Initially I was getting from a Gauge metric that have on our handlers. But now once we are injecting attribute on all requestMetrics otlp object, consequently in all metrics type exist, perhaps now we dont need more metric name, can just extract metric attribute from the first metric found.

Reply:
I fixed it, there is a different approach, by using the type of the metric.